_rules/links-with-identical-names-serve-equivalent-purpose-b20e66.md

```diff
@@ -152,19 +158,183 @@ A set of two SVG `<a>` elements have the same [accessible name][] and link to th
 </svg>
 ```
 
+#### Passed Example 10
+
+A set of one SVG `a` element and one HTML `a` element that have the same [accessible name][] and link to the [same resource][].
+
+```html
+<a href="http://facebook.com">Follow us"</a>
+
+<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+	<a href="http://facebook.com" aria-label="Follow us">
+		<circle cx="50" cy="40" r="35" />
+	</a>
+</svg>
+```
+
+#### Passed Example 11
+
+All three links have the same [accessible name][]. The second link ("from the light") is only part of the [light tree][]. When the [shadow tree][] is attached to `host` and flattened, this link is overwritten and therefore not part of the [flat tree][] (work in progress). Hence, only the first and third link are considered by this rule and they both point to the [same resource][].
+
+```html
+<!DOCTYPE html>
+<html>
+	<head>
+		<title>Links in the shadow</title>
+	</head>
+
+	<body>
+		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a>
+		all the time.
+
+		<div id="host">
+			<span
+				><a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
+					>Contact us</a
+				>
+				from the light.</span
+			>
+		</div>
+
+		<script>
+			const host = document.getElementById('host')
+			const shadowRoot = host.attachShadow({ mode: 'open' })
+
+			shadowRoot.innerHTML =
+				'<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a> from the shadow.'
+		</script>
+	</body>
+</html>
+```
+
+#### Passed Example 12
+
+The [shadow tree][] contains only a default [slot][] (whose [name][] is the empty string). This [slot][] is filled by the third link ("from the slot") and the second one ("from the light") does not appear in the [flat tree][] (work in progress). Therefore, the rule passes.
+
+```html
+<!DOCTYPE html>
+<html>
+	<head>
+		<title>Links in the shadow</title>
+	</head>
+
+	<body>
+		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a>
+		all the time.
+
+		<div id="host">
+			<span slot="slot"
+				><a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
+					>Contact us</a
+				>
+				from the light.</span
+			>
+			<span
+				><a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html"
+					>Contact us</a
+				>
+				from the slot.</span
+			>
+		</div>
+
+		<script>
+			const host = document.getElementById('host')
+			const shadowRoot = host.attachShadow({ mode: 'open' })
+
+			shadowRoot.innerHTML = '<slot></slot>'
+		</script>
+	</body>
+</html>
+```
+
+#### Passed Example 13
```

Comment:
The purpose of these examples is to show what the rule is about, not to check whether or not shadow DOM is correctly implemented. You can potentially add that to any rule. That's just creates a lot of duplicate tests. It's also kind of an anti-pattern to test more than what you are actually trying to show. The only thing this rule needs to verify is that shadow DOM was considered. Not that the shadow DOM implementation is correct. Please remove passed examples 12 - 14.

Reply:
I do not fully understand…

I mean, in July I made the point that many rules have examples that are unrelated to the rule but only duplicate tests with a lot of variation on how to got an accessible name. See #717 for the PR.

Back then, you dismissed the PR in https://github.com/act-rules/act-rules.github.io/pull/717#issuecomment-516377867 by stating, among other thing, that we do not care how or whether implementors have an implementation of "accessible name" and thus should keep all the examples in the rules to check that the implementation of that rule is using a correct way of getting it.

I do not understand how this argument does apply to accessible name but not to the shadow DOM. The current rules have very few shadow DOM examples and thus do not check if implementation correctly handle them, including all the subtleties of slots of stuff. 

Why would it be OK to only verify that shadow DOM was considered, without verifying that it is correct; while it was KO to only verify that accessible name was considered without verifying that it is correct (_e.g._ having only a `aria-label` example and no `aria-labelledby` example)?

I agree that shadow trees could potentially be added to many rules (not all rules actually care about shadow trees). In the exact same way that 4 or 5 different possibilities of giving an accessible name can be added to many rules. I agree that it clutters the rules with useless examples that do not illustrate the rule at all, this was exactly my point in #717.